client/proxy.go

```diff
@@ -0,0 +1,1055 @@
+
+	// TODO switch on egress.Protocol
+	endpoint := qdr.TcpEndpoint{
+		Name:    proxyName + "-egress-" + egress.Address,
+		Host:    egress.Host,
+		Port:    egress.Port,
+		Address: egress.Address,
+	}
+	proxyConfig.AddTcpConnector(endpoint)
+
+	// should this be update or write to config map
+	proxyConfig.UpdateConfigMap(configmap)
+	_, err = cli.KubeClient.CoreV1().ConfigMaps(cli.GetNamespace()).Update(configmap)
+	if err != nil {
+		return fmt.Errorf("Failed to update proxy configmap: %w", err)
+	}
+
+	if isActive(proxyName) {
+		mc, err := qdr.MarshalRouterConfig(*proxyConfig)
+		if err != nil {
+			return fmt.Errorf("Failed to marshall router config: %w", err)
+		}
+
+		err = ioutil.WriteFile(proxyDir+"/config/qdrouterd.json", []byte(mc), 0644)
+		if err != nil {
+			return fmt.Errorf("Failed to write config file: %w", err)
+		}
+
+		url, err := ioutil.ReadFile(proxyDir + "/config/url.txt")
+		agent, err := qdr.Connect(string(url), nil)
+		if err != nil {
+			return fmt.Errorf("Agent error adding tcp connector: %w", err)
+		}
+		record := map[string]interface{}{}
+		if err = convert(endpoint, &record); err != nil {
+			return fmt.Errorf("Failed to convert record: %w", err)
+		}
+		if err = agent.Create("org.apache.qpid.dispatch.tcpConnector", endpoint.Name, record); err != nil {
+			return fmt.Errorf("Error adding tcp connector : %w", err)
+		}
+		agent.Close()
+	}
+	return nil
+}
+
+func (cli *VanClient) ProxyUnbind(ctx context.Context, proxyName string, address string) error {
+	proxyDir := getDataHome() + "/skupper/" + proxyName
+
+	_, err := getRootObject(cli)
+	if err != nil {
+		return fmt.Errorf("Skupper not initialized in %s", cli.Namespace)
+	}
+
+	configmap, err := kube.GetConfigMap(proxyPrefix+proxyName, cli.GetNamespace(), cli.KubeClient)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve proxy configmap: %w", err)
+	}
+	proxyConfig, err := qdr.GetRouterConfigFromConfigMap(configmap)
+
+	deleted, _ := proxyConfig.RemoveTcpConnector(proxyName + "-egress-" + address)
+
+	// should this be update or write to config map
+	proxyConfig.UpdateConfigMap(configmap)
+	_, err = cli.KubeClient.CoreV1().ConfigMaps(cli.GetNamespace()).Update(configmap)
+	if err != nil {
+		return fmt.Errorf("Failed to update proxy configmap: %w", err)
+	}
+
+	if isActive(proxyName) && deleted {
+		mc, err := qdr.MarshalRouterConfig(*proxyConfig)
+		if err != nil {
+			return fmt.Errorf("Failed to marshall router config: %w", err)
+		}
+
+		err = ioutil.WriteFile(proxyDir+"/config/qdrouterd.json", []byte(mc), 0644)
+		if err != nil {
+			return fmt.Errorf("Failed to write config file: %w", err)
+		}
+
+		url, err := ioutil.ReadFile(proxyDir + "/config/url.txt")
+		agent, err := qdr.Connect(string(url), nil)
+		if err != nil {
+			return fmt.Errorf("qdr agent error: %w", err)
+		}
+		if err = agent.Delete("org.apache.qpid.dispatch.tcpConnector", proxyName+"-egress-"+address); err != nil {
+			return fmt.Errorf("Error removing tcp connector : %w", err)
```

Comment:
Either close the connection on error, or have a `defer agent.Close()` after successful connect?